Msg.WriteTo panics with nil pointer dereference when underlying writer returns an error

## Description

`Msg.WriteTo` panics with a nil pointer dereference instead of returning an error when the underlying `io.Writer` fails during a multipart message write.

## Root cause

In `msgwriter.go`, `writeBody` (line ~518) sets `writer = mw.partWriter` when `mw.depth > 0`. However, if a preceding call to `newPart` (line 397–398) failed because `CreatePart` returned an error (due to the underlying writer failing), `mw.partWriter` is set to `nil` while the error is stored in `mw.err`.

`writeBody` does **not** check `mw.err` before using `writer`, so it proceeds to call `io.Copy(writer, &writeBuffer)` (line 566) with a nil `writer`, which panics inside `bytes.(*Buffer).WriteTo`:

```
panic: runtime error: invalid memory address or nil pointer dereference

goroutine ... [running]:
bytes.(*Buffer).WriteTo(0xc..., {0x0, 0x0})
        bytes/buffer.go:279 +0xf1
io.copyBuffer({0x0, 0x0}, {0x..., 0xc...}, {0x0, 0x0, 0x0})
        io/io.go:411 +0xd4
github.com/wneessen/go-mail.(*msgWriter).writeBody(...)
        msgwriter.go:566
github.com/wneessen/go-mail.(*msgWriter).writePart(...)
        msgwriter.go:438
github.com/wneessen/go-mail.(*msgWriter).writeMsg(...)
        msgwriter.go:182
github.com/wneessen/go-mail.(*Msg).WriteTo(...)
        msg.go:2444
```

## Steps to reproduce

```go
package main

import (
	"fmt"

	gomail "github.com/wneessen/go-mail"
)

type failWriter struct{}

func (f *failWriter) Write(_ []byte) (int, error) {
	return 0, fmt.Errorf("simulated write failure")
}

func main() {
	msg := gomail.NewMsg()
	_ = msg.From("sender@example.com")
	_ = msg.To("recipient@example.com")
	msg.Subject("Test")
	msg.SetBodyString(gomail.TypeTextPlain, "plain text")
	msg.AddAlternativeString(gomail.TypeTextHTML, "<p>html</p>")

	// This panics instead of returning an error:
	_, err := msg.WriteTo(&failWriter{})
	fmt.Println("error:", err)
}
```

## Expected behaviour

`WriteTo` should return the write error (already captured in `mw.err`) instead of panicking.

## Suggested fix

Add an early return in `writeBody` (and/or `writePart`) if `mw.err` is already set, or add a nil check on `writer` before calling `io.Copy`:

```go
func (mw *msgWriter) writeBody(writeFunc func(io.Writer) (int64, error), encoding Encoding) {
	var writer io.Writer
	// ...
	if mw.depth > 0 {
		writer = mw.partWriter
	}
+	if writer == nil {
+		return
+	}
	// ...
}
```

## Environment

- **go-mail version:** v0.7.2
- **Go version:** go1.26.2 linux/amd64

--- a/msg_test.go
+++ b/msg_test.go
@@ -6912,6 +6912,26 @@ func TestMsg_WriteTo(t *testing.T) {
 	})
 }
 
+func TestMsg_WriteTo_FailingWriter(t *testing.T) {
+	t.Run("WriteTo with failing writer does not panic", func(t *testing.T) {
+		message := NewMsg()
+		if err := message.From(TestSenderValid); err != nil {
+			t.Fatalf("failed to set from address: %s", err)
+		}
+		if err := message.To(TestRcptValid); err != nil {
+			t.Fatalf("failed to set to address: %s", err)
+		}
+		message.Subject("Test panic fix")
+		message.SetBodyString(TypeTextPlain, "plain text")
+		message.AddAlternativeString(TypeTextHTML, "<p>html</p>")
+
+		_, err := message.WriteTo(failReadWriteSeekCloser{})
+		if err == nil {
+			t.Error("expected error when writing to a failing writer, got nil")
+		}
+	})
+}
+
 func TestMsg_WriteToFile(t *testing.T) {
 	t.Run("WriteToFile with normal mail parts", func(t *testing.T) {
 		tempfile, err := os.CreateTemp("", "testmail.*.eml")

--- a/msgwriter.go
+++ b/msgwriter.go
@@ -525,6 +525,9 @@ func (mw *msgWriter) writeBody(writeFunc func(io.Writer) (int64, error), encodin
 	if mw.depth > 0 {
 		writer = mw.partWriter
 	}
+	if writer == nil {
+		return
+	}
 	writeBuffer := bytes.Buffer{}
 	lineBreaker := base64LineBreaker{}
 	lineBreaker.out = &writeBuffer

--- a/msgwriter_test.go
+++ b/msgwriter_test.go
@@ -763,6 +763,18 @@ func TestMsgWriter_writeBody(t *testing.T) {
 			t.Errorf("expected error: bodyWriter function: intentional write failure, got: %s", msgwriter.err)
 		}
 	})
+	t.Run("writeBody with nil partWriter does not panic", func(t *testing.T) {
+		mw := &msgWriter{
+			charset: CharsetUTF8,
+			encoder: getEncoder(EncodingQP),
+			depth:   1,
+		}
+		mw.partWriter = nil
+		writeFunc := func(w io.Writer) (int64, error) {
+			return 0, nil
+		}
+		mw.writeBody(writeFunc, EncodingQP)
+	})
 }
 
 func TestMsgWriter_sanitizeFilename(t *testing.T) {

